core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -499,6 +499,10 @@ sealed abstract class Parser[+A] extends Parser0[A] {
     if (min == 0) rep0
     else this.repAs(min)
 
+  def rep0(min: Int, max: Int): Parser0[List[A]] =
+    if (min == 0) rep0
```

Comment:
Isn't this discarding max? Can we add a test that would have caught this?

Reply:
Yes, it was. The property test that tests p.rep(min, max) results in no more than max results didn't catch this, even locally after running it a bunch of times in a row. There is an explicit test for this now.

Since the property test didn't catch it, I'm concerned it just eats CPU cycles and feedback loop latency.